launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -172,97 +177,106 @@ public void doPost_ReturnsCorrectUrl() throws IOException {
     assertEquals(1, ds.prepare(new Query(USER_FEED)).countEntities(withLimit(10)));
 
     Query query = new Query(USER_FEED);
-    Entity entity = ds.prepare(query).asSingleEntity();
+    PreparedQuery preparedQuery = ds.prepare(query);
+    Entity desiredEntity = preparedQuery.asSingleEntity();
+
+    String expectedXmlString = RSS.toXmlString(TEST_RSS_FEED);
+    assertEquals(expectedXmlString, desiredEntity.getProperty(XML_STRING).toString());
 
-    assertEquals(TEST_PODCAST_TITLE, entity.getProperty(PODCAST_TITLE).toString());
-    assertEquals(TEST_MP3_LINK, entity.getProperty(MP3_LINK).toString());
-    RSS rssFeed = new RSS(TEST_PODCAST_TITLE, TEST_MP3_LINK);
-    TEST_XML_STRING = RSS.toXmlString(rssFeed);
-    assertEquals(TEST_XML_STRING, entity.getProperty(XML_STRING).toString());
+    String test_xml_string = RSS.toXmlString(TEST_RSS_FEED);
+    assertEquals(test_xml_string, desiredEntity.getProperty(XML_STRING).toString());
 
-    String id = Long.toString(entity.getKey().getId());
+    String id = KeyFactory.keyToString(desiredEntity.getKey());
     String rssLink = BASE_URL + id;
+    System.out.println("THIS IS THE EXPECTED LINK: " + rssLink);
+    System.out.println("THIS IS THE ACTUAL   LINK: " + stringWriter.toString());
+    System.out.println("length 1=" + rssLink.length());
+    System.out.println("length 2=" + stringWriter.toString().length());
 
-    verify(response, atLeast(1)).setContentType("text/html");
-    assertEquals(rssLink, stringWriter.toString());
+    verify(response, times(1)).setContentType("text/html");
+    assertEquals(0, rssLink.compareTo(stringWriter.toString()));
   }
 
   /**
-   * Expects doPost() to throw an IOException when the title field is empty.
+   * Expects doPost() to throw an IllegalArgumentException when the title field is
+   * empty.
    */
   @Test
   public void doPost_FormInputEmptyTitle_ThrowsErrorMessage() throws IOException {
     DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
     when(request.getParameter(PODCAST_TITLE)).thenReturn("");
     when(request.getParameter(MP3_LINK)).thenReturn(TEST_MP3_LINK);
 
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("No Title inputted, please try again.");
     servlet.doPost(request, response);
 
     assertEquals(1, ds.prepare(new Query(USER_FEED)).countEntities(withLimit(10)));
 
     Query query = new Query(USER_FEED);
     PreparedQuery results = ds.prepare(query);
     Entity entity = ds.prepare(query).asSingleEntity();
```

Comment:
do we need this portion? doPost should be the one throwing the exception, and this doesn't do anything. did you mean to test Entity somehow?

same with other tests

Reply:
Forgot to delete these